### PR TITLE
Automate some of the previously manual mac tests

### PR DESCRIPTION
**Pull request by timrisi**
Automates:

System.ServiceModel and XM 4.5
•	Verify that user is able to run the following command and get expected result.
•	Verify that user is able to build and run attached "ServiceModel_Test" test sample and get expected result.
•	Verify that user is able to build and run attached "Protobuf_Test" test sample and get expected result.
XM 4.5 Full Framework Under Unified
•	Verify that when user build a project targeting the new “Xamarin.Mac .NET 4.5 Framework", should see the following in the build log under Target _CompileToNative.
All Test Cases
•	Verify that user is able to build the application after adding the reference to OpenTk.
Additional Framework Support
•	Verify that user is able to build and deploy the Xamarin.Mac application using "Xamarin.Mac .Net 4.5 Framework".
Basic PCL Test
Verify that user gets correct result in application output, on running attached sample in release mode with correct provisioning profile

### CODE REVIEW

**Comment by spouliot** [comment]
Good work but this needs some discussions before merging.

**Comment by timrisi** [comment]
Most or all can nuke the UIs.  Several of them used them originally because they were displaying NSAlerts with the result you were looking for to verify it was working, I killed the alerts in favor of writing out the result, so shouldn't be a a big deal to nuke the xibs.

File: tests/xammac_tests/Unified45.cs [hunk 1]
```diff
@@ -0,0 +1,65 @@
+﻿using System;
+using System.IO;
+using NUnit.Framework;
+using Xamarin.MMP.Tests;
+
+namespace MonoTouchFixtures.Net45 {
+	[TestFixture]
+	public class Unified45 {
+
```
Comment (chamons) [inline tests/xammac_tests/Unified45.cs]: @timrisi Remove this extra space?

**Comment by timrisi** [comment]
Hrmm, so looks like the test projects are failing to pull in the nuget packages that several of them use (presumably needing to restore the packages before being able to build successfully).  @chamons @spouliot Is there any way to restore packages in the testing process?

**Comment by rolfbjarne** [comment]
The mtouch tests call nuget restore: https://github.com/xamarin/xamarin-macios/blob/master/tests/mtouch/Makefile#L32

**Pull request merged**